src/common/transformations/src/transformations/flush_fp32_subnormals_to_zero.cpp

```diff
@@ -36,14 +36,28 @@ ov::pass::FlushFP32SubnormalsToZero::FlushFP32SubnormalsToZero() {
         bool has_subnormals = false;
         for (size_t i = 0; i < size; ++i) {
             if (fpclassify(std::abs(data[i])) == FP_SUBNORMAL) {
-                data[i] = 0.0f;
                 has_subnormals = true;
+                break;
             }
         }
-        if (has_subnormals)
-            return true;
+        if (!has_subnormals)
+            return false;
+
+        auto new_constant = std::make_shared<ov::opset8::Constant>(ov::element::f32, node->get_shape());
```

Comment:
I believe you need to use type of the old constant
auto new_constant = std::make_shared<ov::opset8::Constant>(old_constant->get_element_type(), node->get_shape());

Reply:
I see we have a check in the transformation
        if (node->get_output_element_type(0) != element::f32)
            return false;